test/jdk/javax/swing/JTable/JTableRightAlignmentTest.java

```diff
@@ -0,0 +1,200 @@
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.AbstractTableModel;
+import javax.swing.UIManager;
+
+import javax.imageio.ImageIO;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * @test
+ * @key headful
+ * @bug 5108458
+ * @summary Test to check Right alignment of JTable data
+ * (Fix affects all L&F, test verifies only Metal L&F)
+ * @run main JTableRightAlignmentTest
+ */
+
+public class JTableRightAlignmentTest {
+    static JFrame frame;
+    static CustomTable table;
+    static boolean passed = true;
+    static String failureString;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        Robot robot = new Robot();
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                frame = new JFrame("Test JTable");
+
+                JPanel panel = new JPanel(new GridBagLayout());
+                frame.setContentPane(panel);
+                table = new CustomTable();
+                panel.add(new JScrollPane(table.table),
+                        new GridBagConstraints(0, 0, -1, -1, 1.0, 1.0,
+                                GridBagConstraints.PAGE_START, GridBagConstraints.BOTH,
+                                new Insets(2, 2, 2, 2), 0, 0));
+                frame.pack();
+                frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+                frame.applyComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+                frame.setVisible(true);
+            });
+
+            robot.waitForIdle();
+            robot.delay(1000);
+            SwingUtilities.invokeAndWait(() -> {
+                int maxHeight = (int) (((double) table.table.getTableHeader().getHeight())
+                        + ((double) table.table.getHeight()));
+                int yPos = table.table.getTableHeader().getLocationOnScreen().y;
+                int xPos = table.table.getLocationOnScreen().x + table.table.getWidth() -
+                        table.table.getTableHeader().getColumnModel()
+                                .getColumn(2)
+                                .getWidth() - 1;
+                Color expectedRGB = robot.getPixelColor(xPos, yPos);
+                for (int y = yPos; y < yPos + maxHeight; y++) {
+                    if (expectedRGB.getRGB() != robot.getPixelColor(xPos, y).getRGB()) {
+                        BufferedImage failImage = robot.createScreenCapture(
+                                new Rectangle(xPos, yPos, 3, maxHeight));
+                        saveImage(failImage, "failureImage.png");
+                        passed = false;
+                        failureString = "Test Failed at <" + xPos + ", " + y + ">";
```

Comment:
`failureString == null` implies `passed` has the value `true` — one variable / field is enough.